src/js/utils/ConditionalInterval.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+class ConditionalInterval
+{
+    constructor(callback, milliseconds, condition)
+    {
+        this._interval = setInterval(() => {
```

Comment:
This should probably use `GUI.interval_add`, for consistency, and to make sure it is cancelled on disconnect.

Also not 100% sure that this needs its own class, the use case seems to be quite narrow.